test/FlowChartSpec.hs

```diff
@@ -0,0 +1,16 @@
+module FlowChartSpec
+  ( spec,
+  )
+where
+
+import Test.Hspec
+import Text.Megaparsec
+import FlowChart
+import Algebra.Graph.Labelled as LG (Graph(..))
+
+spec :: Spec
+spec = do
+  describe "FlowChart" $ do
+    context "FlowChart" $ do
+      it "parse single node" $ do
+        parseMaybe pDiagram "flowchart TD\n  id" `shouldBe` Just (FlowChart {orientation = TD, graph = Connect Nothing (Connect Nothing (Vertex (Node {nodeId = NodeId {unNodeId = "id"}, nodeShape = Nothing, nodeLabel = Nothing})) Empty) Empty})
```

Comment:
1. You can use [neat-interpolation](https://hackage.haskell.org/package/neat-interpolation-0.3.2.1/docs/NeatInterpolation.html) to lay out the flowchart code in mulitple lines for legibility
2. The whole `Connect Nothing (Connect ... ` is a bit jarring to read. We can do better I think?

One idea basically is to write a variation of `shouldBe` -- calling it `shouldHaveVertexAndEdges` (or `shouldBeGraph`) for instance -- that takes a list of vertexes and a list of edge pairs, and then checks if the graph matches that. 

```haskell
let diagram = 
      [text|
        flowchart TD
          id
      |]
parseMaybe pDiagram diagram 
  `shouldBeGraph` (["id"], [])`
```

Also, you can get rid of the `context` thing.

Reply:
Yes, that will work too. But you'd also need `vertexSet` to compare disconnected vertices if any.